pkg/controllers/scheduler/scheduler.go

```diff
@@ -95,6 +95,14 @@ type Scheduler struct {
 	logger  klog.Logger
 }
 
+func (s *Scheduler) IsControllerReady() bool {
+	return s.federatedObjectSynced() &&
+		s.clusterPropagationPolicySynced() &&
```

Comment:
This looks wrong... Can we extract the `cachesSynced` slice in `Run` into a `HasSynced` function and call that instead?